opencog/nlp/sureal/SuRealPMCB.cc

```diff
@@ -426,17 +428,82 @@ bool SuRealPMCB::grounding(const std::map<Handle, Handle> &var_soln, const std::
         shrinked_soln[kv.first] = kv.second;
     }
 
-    // store the solution; all common InterpretationNode are solutions for this
-    // grounding, so store the solution for each InterpretationNode
-    for (auto n : qItprNode)
+    // if there are more than one common InterpretationNodes at this point,
+    // we should also have more than one SetLinks here, so examine them one by one
+    for (Handle hSetLink : qSolnSetLinks)
```

Comment:
Doesn't seem right... qItprNode is shrinking on each iteration, but qSolnSetLinks is growing...